bepasty_cli/cli.py

```diff
@@ -30,6 +31,18 @@
 LIFETIME_MAPPING = dict(zip(LIFETIME_CHOICES, LIFETIME_NAMES))
 
 
+class LifetimeParamType(click.ParamType):
+    name = 'lifetime'
+    lifetime_regex = re.compile(r'|'.join(r'^([1-9]\d*?) ?({})$'.format(choice) for choice in LIFETIME_CHOICES))
+
+    def convert(self, value, param, ctx):
+        m = self.lifetime_regex.match(value)
+        if m:
+            return [x for x in m.groups() if x]
```

Comment:
m.group() returns `None` for each unmatched regex-group (int, 'min', int, 'd', int, 'day', int, 'm', int, 'month',...)
`-m 2d` → [None, None, 2, 'd', None, None, None, ...]